src/compile/axis/parse.ts

```diff
@@ -120,6 +120,8 @@ function getSpecifiedOrDefaultValue(property: keyof VgAxis, specifiedAxis: Axis,
   switch (property) {
     case 'labels':
       return isGridAxis ? false : specifiedAxis[property];
+    case 'labelOverlap':
+      return rules.labelOverlap(fieldDef, specifiedAxis, channel, isGridAxis);  // TODO: scaleType
```

Comment:
Also, you can easily getScaleComponent and read the type here to handle overlapping for discrete and log scales.